pkg/cloudprovider/aws/fleet/instance.go

```diff
@@ -84,6 +84,9 @@ func (p *InstanceProvider) Create(ctx context.Context,
 }
 
 func (p *InstanceProvider) Terminate(ctx context.Context, nodes []*v1.Node) error {
+	if len(nodes) == 0 {
```

Comment:
Sorry if I missed the comment. Why is this necessary?

Reply:
Nvm -- this is clearly a stupid EC2 impl detail. Why EC2 doesn't happily say OK to zero deletions is beyond me.